data_pipelines_cli/airbyte_utils.py

```diff
@@ -48,42 +55,68 @@ def create_update_connections(self) -> None:
             [task.update(self.env_replacer(task)) for task in self.airbyte_config["tasks"]]
             self.update_file(self.airbyte_config)
 
+    def get_workspace_id(self) -> str:
+        workspaces = self.request_handler("workspaces/list").get("workspaces")
+        if not workspaces:
+            raise AirbyteNoWorkspaceConfiguredError(
+                f"No workspaces found in {self.airbyte_url} instance."
+            )
+
+        return workspaces[0].get("workspaceId")
+
     def create_update_connection(self, connection_config: Dict[str, Any]) -> Any:
+        def configs_equal(
+            conf_a: Dict[str, Any], conf_b: Dict[str, Any], equality_fields: Iterable[str]
+        ) -> bool:
+            conn_a = {k: v for k, v in conf_a.items() if k in equality_fields}
+            conn_b = {k: v for k, v in conf_b.items() if k in equality_fields}
+            return conn_a == conn_b
+
         connection_config_copy = copy.deepcopy(connection_config)
         response_search = self.request_handler(
-            "connections/search",
-            {
-                "sourceId": connection_config_copy["sourceId"],
-                "destinationId": connection_config_copy["destinationId"],
-                "namespaceDefinition": connection_config_copy["namespaceDefinition"],
-                "namespaceFormat": connection_config_copy["namespaceFormat"],
-            },
+            "connections/list",
+            data={"workspaceId": self.get_workspace_id()},
```

Comment:
Does connection in connection_config have a workspace? I think it should and that one should be used to verify. I think there will be up to a few connections per project so we could process them one by one instead of everything. It should simplify a little bit.

Reply:
@p-pekala Just to make sure I understand the suggestion correctly, you meant that we should include `workspaceId` as part of the each connection entry like below?

```yaml
airbyte_connection_id: airbyte_connection_id
airbyte_url: <airbye_url>
connections:
  conn_1:
    ...
    workspaceId: 0c5589a1-40ab-4ea7-b842-9b8276554b3e
    name: conn_1
    ...
  conn_2:
    ...
    workspaceId: 9930e98e-7138-450d-b48f-196c6373ba5c
    name: conn_2
    ...
```

Reply:
@p-pekala `workspace_id` is now added to `airbyte.yml` config file.